controllers/userController.js

```diff
@@ -0,0 +1,31 @@
+const { userService } = require("../services");
+
+const signUp = async (req, res) => {
+    try {
+        const { type_id, name, email, password, account } = req.body;
+
+        if (type_id === 1) {
+            if (!name || !email || !password) {
+                const error = new Error("KEY_ERROR: Missing required fields: name, email, password.");
+                error.statusCode = 400;
+                throw error;
+            }
+        } else {
+            if (!account) {
+                const error = new Error("KEY_ERROR: Missing required field.");
+                error.statusCode = 400;
+                throw error;
+            }
+        }
```

Comment:
타입이 1이 아닐 때도 이름이나 비밀번호는 필요한거 아닌가요?
```suggestion
          if (!name || !email || !password) {
              const error = new Error("KEY_ERROR: Missing required fields: name, email, password.");
              error.statusCode = 400;
              throw error;
          }
        }
        
        if (type_id !==1 && !account) {
            const error = new Error("KEY_ERROR: Missing required field.");
            error.statusCode = 400;
            throw error;
        }
     }
```